asn1/CommonLedgerProtocol.asn

```diff
@@ -0,0 +1,109 @@
+CommonLedgerProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Timestamp
+    FROM InterledgerTypes
+
+    InterledgerProtocolError
+    FROM InterledgerProtocol
+;
+
+-- Zero or more protocolData entries, each with a unique
+-- protocolName, describe protocol-specific information about
+-- unpaid, paid, and conditionally paid requests and results
+-- sent over the CommonLedgerProtocol.
+-- The protocolName of each entry describes the protocol for
+-- interpreting the data. Peers need to establish beforehand
+-- which protocols each understand, and which numbers they
+-- use, which can be agreed ad-hoc,except that protocolName 'ilp'
+-- is reserved for the Interledger protocol.
+-- When multiple protocolData items are sent in a request,
+-- only the first one will be decisive in triggering the
+-- response, the rest of the protocolData items are considered
+-- side protocols that piggyback on the first one.
+-- The contentType byte can be used to allow debug tools to
+-- display the contents of the protocolData without understanding
+-- it; set to 0 for `application/octet-string`, 1 for
+-- `text/plain; charset=utf-8`, and 2 for `application/json`.
+
+ProtocolData ::= SEQUENCE OF SEQUENCE {
+  protocolName IA5String,
+  contentType UInt8,
+  data OCTET STRING
+}
+
+-- Ack, Response, and Error are the response types.
+-- When using these in a CLP packet, the requestId should match
+-- the requestId of the request they respond to.
+
+Ack ::= ProtocolData
+Response ::= ProtocolData
+Error ::= SEQUENCE {
+  rejectionReason InterledgerProtocolError,
+  --
+  protocolData ProtocolData
+}
+
+-- Prepare, Fulfill, Reject, and Message are the request types.
+-- Each request should have a unique requestId.
```

Comment:
I agree that in an asymmetric trustline, you often want to send messages without caring about response. But Adrian raised a good point that if all notifications have ID 0 it's hard to do any logging with them. Furthermore, in a client/server relationship the server _never_ cares whether the client responds, so it can be agreed at establishment that the client doesn't respond, rather than on a per-message basis